### ISSUE TITLE
E203 false positives for slices in format strings

### ISSUE THREAD

**Issue by sciyoshi** [astral-sh/ruff#10278]
Sample code:

```python
a = f"{1[1 - 2 :]}"
b = 1[1 - 2 :]
```

When running with `ruff --select E203 --preview`, the first line triggers E203:

```
test.py:1:15: E203 [*] Whitespace before ':'
  |
1 | a = f"{1[1 - 2 :]}"
  |               ^ E203
  |
  = help: Remove whitespace before ':'
```

The same slice on the second line which is outside of a format string does not trigger E203.

**Comment by charliermarsh** [comment]
Thanks for the PR as always.
